databricks-sdk-java/src/main/java/com/databricks/sdk/AccountClient.java

```diff
@@ -453,159 +453,293 @@ public WorkspacesAPI workspaces() {
     return workspacesAPI;
   }
 
-  /** Override AccountAccessControlAPI with mock */
+  /** Replace the default AccountAccessControlService with a custom implementation. */
   public AccountClient withAccessControlImpl(AccountAccessControlService accountAccessControl) {
-    accessControlAPI = new AccountAccessControlAPI(accountAccessControl);
+    return this.withAccessControlAPI(new AccountAccessControlAPI(accountAccessControl));
+  }
+
+  /** Replace the default AccountAccessControlAPI with a custom implementation. */
```

Comment:
LGTM. Curious -- are we expecting customers to have custom implementation for these APIs? What is the expected use case for this?